File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/member/model/Member.java
/*
 * Copyright © 2015 The Gravitee team (http://gravitee.io)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package io.gravitee.apim.core.member.model;

import io.gravitee.rest.api.model.MembershipMemberType;
import io.gravitee.rest.api.model.MembershipReferenceType;
import io.gravitee.rest.api.model.permissions.RoleScope;
import io.gravitee.rest.api.model.permissions.SystemRole;
import java.util.Date;
import java.util.List;
import java.util.Map;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @author Kamiel Ahmadpour (kamiel.ahmadpour at graviteesource.com)
 * @author GraviteeSource Team
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
public class Member {

    private String id;
    private String displayName;
    private String email;
    private MembershipMemberType type;
    private MembershipReferenceType referenceType;
    private String referenceId;
    private List<Role> roles;
    private Map<String, char[]> permissions;
    private Date createdAt;
    private Date updatedAt;

    @AllArgsConstructor
    @NoArgsConstructor
    @Data
    public static class Role {

        private String id;
        private String name;
        private String description;
        private RoleScope scope;
        private boolean defaultRole;
        private boolean system;
        private Map<String, char[]> permissions;

        public boolean isPrimaryOwner() {
            return scope == RoleScope.APPLICATION && SystemRole.PRIMARY_OWNER.name().equals(name);
        }
    }
}

File: gravitee-apim-portal-webui-next/src/entities/member/member.ts
/*
 * Copyright (C) 2024 The Gravitee team (http://gravitee.io)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { User } from '../user/user';

export interface Member {
  id?: string;
  user?: User;
  created_at?: string;
  updated_at?: string;
  role?: string;
}

export interface MemberInput {
  user?: string;
  reference?: string;
  role: string;
}

export interface TransferOwnershipInput {
  new_primary_owner_id: string;
  new_primary_owner_reference?: string;
  primary_owner_newrole: string;
}

export interface MembersResponse {
  data?: Member[];
  metadata?: {
    pagination?: {
      current_page?: number;
      total?: number;
      size?: number;
    };
  };
  links?: unknown;
}

/** Filters supported by member search APIs (e.g. application members `_search`). */
export interface MemberSearchFilters {
  displayName?: string;
}
